code/game/objects/items/stacks/golem_food/golem_status_effects.dm

```diff
@@ -213,25 +323,141 @@
 	owner.Beam(target, icon_state = "lightning[rand(1,12)]", time = 0.5 SECONDS)
 	playsound(owner, 'sound/effects/magic/lightningshock.ogg', vol = 50, vary = TRUE)
 
-/// Makes you spaceproof
+/// Makes you tenacious, but not invincible
 /datum/status_effect/golem/plasteel
 	overlay_state_prefix = "iron"
 	mineral_name = "plasteel"
-	applied_fluff = "Plasteel plates seal you tight. You feel insulated!"
+	applied_fluff = "Plasteel plates reinforce your body and your will!"
 	alert_icon_state = "sheet-plasteel"
-	alert_desc = "You are sealed against the cold, and against low pressure environments."
+	alert_desc = "You won't take damage while in crit, and you're resistant (but not immune) to the cold-welding effect of low pressure. Healing out of crit is another problem..."
+	// keep track of an outline for our pressure alerts so we can make sure it gets removed
+	var/active_outline = null
+	// keep track of the alert so we can actually do that
+	var/atom/movable/screen/alert/golem_coldwelding/active_pressure_alert = null
+	// and finally, keep track of if we've faded in the outline yet so we don't needlessly animate it
+	var/effect_faded_in = FALSE
 
 /datum/status_effect/golem/plasteel/on_apply()
 	. = ..()
 	if (!.)
 		return FALSE
-	owner.add_traits(list(TRAIT_RESISTLOWPRESSURE, TRAIT_RESISTCOLD), TRAIT_STATUS_EFFECT(id))
+	owner.add_traits(list(TRAIT_NOCRITDAMAGE), TRAIT_STATUS_EFFECT(id))
+	start_pressure_slow_resistance_mitigation()
 	return TRUE
 
 /datum/status_effect/golem/plasteel/on_remove()
-	owner.remove_traits(list(TRAIT_RESISTLOWPRESSURE, TRAIT_RESISTCOLD), TRAIT_STATUS_EFFECT(id))
+	owner.remove_traits(list(TRAIT_NOCRITDAMAGE), TRAIT_STATUS_EFFECT(id))
+	stop_pressure_slow_resistance_mitigation()
+	return ..()
+
+/datum/status_effect/golem/plasteel/proc/start_pressure_slow_resistance_mitigation()
```

Comment:
This implementation is a bit messy, to make a suggestion as to how to reorganise it:

- Keep most of this code in `/datum/movespeed_modifier/status_effect/golem_coldwelding`
- Inside the coldwelding debuff, register signals for addition or removal of a trait applied by this status effect and use that to decide how slow the golem should be (if at all). Unfortunately `TRAIT_STATUS_EFFECT(id)` probably won't work because all golem status effects have the same id.
- Also check for that trait on initial application.
- You probably then don't need to care about handling species changes because you can also just do that in the cold welding effect and have it qdel itself.